src/d3a/models/market/two_sided.py

```diff
@@ -253,14 +253,14 @@ def accept_bid_offer_pair(self, bid, offer, clearing_rate, trade_bid_info, selec
                                     seller_id=offer.seller_id)
         return bid_trade, trade
 
-    def match_recommendation(self, recommended_list):
+    def match_recommendations(self, recommended_list: List[Dict]):
```

Comment:
Hmm, isn't is rather
```suggestion
    def match_recommendations(self, recommended_list: List[BidOfferMatch]):
```
? 
In the d3a-interface in `_create_bid_offer_matches` indeed BidOfferMatch objects are appended.

Reply:
It's True I am returning BidOfferMatch after calling the `serializable_dict()` function on it, so these are dicts
I'll update the type hints so it reflects this

Reply:
Nit: since we use types, maybe we don't need to specify that `_list`. But this is really not so important.
```suggestion
    def match_recommendations(self, recommendations: List[BidOfferMatch]):
```

Reply:
> It's True I am returning BidOfferMatch after calling the `serializable_dict()` function on it, so these are dicts
> I'll update the type hints so it reflects this

Ok, i see now. Can you maybe explain why you did that? Isn't calling `record.market_id` as convenient as `record["market_id"]` . One of the advantages of the DataClass, right?

Reply:
True, but I'm sending recommendations through redis or websocket from myco  to d3a, this is why we can't send Dataclass instances